src/Voron/Impl/Scratch/PageFromScratchBuffer.cs

```diff
@@ -16,14 +16,14 @@ public bool Equals(PageFromScratchBuffer x, PageFromScratchBuffer y)
             if (x == y) return true;
             if (x == null || y == null) return false;            
 
-            return x.PositionInScratchBuffer == y.PositionInScratchBuffer && x.NumberOfPages == y.NumberOfPages && x.File == y.File;
+            return x.PositionInScratchBuffer == y.PositionInScratchBuffer && x.Size == y.Size && x.NumberOfPages == y.NumberOfPages && x.File.Number == y.File.Number;
         }
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public int GetHashCode(PageFromScratchBuffer obj)
         {
             int v = Hashing.Combine(obj.NumberOfPages, obj.File.Number);
-            int w = Hashing.Combine(obj.PositionInScratchBuffer.GetHashCode(), obj.PageNumberInDataFile.GetHashCode());
```

Comment:
`PageNumberInDataFile` isn't actually the property of the scratch page itself. It's rather about the content so it shouldn't be considered in hash code calculation. Also note that it was not checked in Equals.